dev-tools/globals.ts

```diff
@@ -538,7 +541,7 @@ interface Move extends Effect, MoveData {
 	zPowered?: boolean
 }
 
-type TemplateAbility = {0: string, 1?: string, H?: string, S?: string}
+type TemplateAbility = {[key: string]: string | undefined, [index: number]: string | undefined, 0: string, 1?: string, H?: string, S?: string}
```

Comment:
Ugh I vetoed this sort of thing. Keep the `@ts-ignore`s everywhere; it's better than wrecking this type and making it worse than useless.

Reply:
What do you mean by "worse than useless"?

Reply:
Its not strict enough to be effective at ensuring that abilities only contain the correct info, see also: https://github.com/Zarel/Pokemon-Showdown/pull/4782#pullrequestreview-143007785

Reply:
Oh. I didn't realize that adding a key signature or w/e it's called was a free pass to any and all properties as long as they're numbers. I guess it never occurred to me they'd make such a huge breaking change as to prevent iterating through objects with a strict list of properties, even if all the properties have the same type, and then suggest a change that ruins the point of defining the type in the first place.

Reply:
TypeScript never allowed iterating objects with a strict list of properties, because TypeScript types are interfaces, not two-way strict types.

We've always dealt with that with `ts-ignore`s everywhere.

Reply:
By which I mean, `{a: 1, b: 'hi'}` satisfies the type `{a: number}`, so iterating through a type with only numbers in no way guarantees that all the properties will be numbers.

Reply:
Blegh.

Reply:
By which I mean, thank you for explaining that, it makes sense to me now.